src/rpc/blockchain.cpp

```diff
@@ -2334,30 +2362,117 @@ UniValue CreateUTXOSnapshot(
         tip = CHECK_NONFATAL(chainstate.m_blockman.LookupBlockIndex(stats.hashBlock));
     }
 
-    LOG_TIME_SECONDS(strprintf("writing UTXO snapshot at height %s (%s) to file %s (via %s)",
-        tip->nHeight, tip->GetBlockHash().ToString(),
-        fs::PathToString(path), fs::PathToString(temppath)));
-
-    SnapshotMetadata metadata{tip->GetBlockHash(), stats.coins_count, tip->nChainTx};
+    switch (format) {
+    case UTXOSnapshotFormat::COMPACT:
+        {
+            LOG_TIME_SECONDS(strprintf("writing UTXO snapshot at height %s (%s) to file %s (via %s)",
+                                    tip->nHeight, tip->GetBlockHash().ToString(),
+                                    fs::PathToString(path), fs::PathToString(temppath)));
+            SnapshotMetadata metadata{tip->GetBlockHash(), stats.coins_count, tip->nChainTx};
+
+            afile << metadata;
+
+            COutPoint key;
+            Coin coin;
+            unsigned int iter{0};
+
+            while (pcursor->Valid()) {
+                if (iter % 5000 == 0) node.rpc_interruption_point();
+                ++iter;
+                if (pcursor->GetKey(key) && pcursor->GetValue(coin)) {
+                    afile << key;
+                    afile << coin;
+                }
 
-    afile << metadata;
+                pcursor->Next();
+            }
 
-    COutPoint key;
-    Coin coin;
-    unsigned int iter{0};
-
-    while (pcursor->Valid()) {
-        if (iter % 5000 == 0) node.rpc_interruption_point();
-        ++iter;
-        if (pcursor->GetKey(key) && pcursor->GetValue(coin)) {
-            afile << key;
-            afile << coin;
+            afile.fclose();
+            break;
         }
+    case UTXOSnapshotFormat::SQLITE:
+        {
+#ifdef USE_SQLITE
+            LOG_TIME_SECONDS(strprintf("writing UTXO snapshot at height %s (%s) to sqlite DB file %s",
+                                    tip->nHeight, tip->GetBlockHash().ToString(), fs::PathToString(path)));
 
-        pcursor->Next();
-    }
+            sqlite3* db;
+            int ret;
+
+            if (!ValidAsCString(fs::PathToString(path))) {
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid path");
+            }
+
+            ret = sqlite3_open(fs::PathToString(path).c_str(), &db);
+            if (ret != SQLITE_OK) {
+                sqlite3_close(db);
+                throw JSONRPCError(RPC_INTERNAL_ERROR, strprintf("Unable to open database file %s", path.u8string()));
+            }
 
-    afile.fclose();
+            // We have no need for rollbacks here, so we can set the `journal_mode` to 'off'. See: https://www.sqlite.org/pragma.html#pragma_journal_mode
+            ret = sqlite3_exec(db, "PRAGMA journal_mode=off;", nullptr, nullptr, nullptr);
+            if (ret != SQLITE_OK) {
+                throw std::runtime_error(strprintf("SQLiteDatabase: Failed to set wal2: %s\n", sqlite3_errstr(ret)));
```

Comment:
I think wal2 is a hangover from a previous version here

Reply:
Indeed, thanks!